### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by janisz**
Added
* all endpoints to `api.yaml` spec.
* links to design docs
* section about project history and future
* section about diagnostics bundle API with diagrams

### CODE REVIEW

File: docs/diagrams/cluster_bundle_creation_diagram.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
+title Cluster Bundle Creation
+
+User ->+ Master: PUT diagnostics/1
+Master ->-User: Started
+
+Master->+Node: PUT node/diagnostics/1234
+Node ->-Master: Started
+
+note right of Node: Node prepares a bunle for itself \n It reads the configuration \n and copy all files to a single ZIP
```
Comment (ArmandGrillet) [inline docs/diagrams/cluster_bundle_creation_diagram.txt]: s/`bunle`/`bundle`

File: README.md [hunk 2]
```diff
@@ -1,28 +1,82 @@
...
-|Enum|Meaning    |
-|----|-----------|
-|  0 | working   |
-|  1 | error     |
-|  3 | unknown   |
+Originally dcos-diagnostics was designed in Master/Agent model. It's running on every DC/OS node.
+
+* Master
+
+Master runs on DC/OS Masters. There is the point of entry to dcos-diagnostics from remote systems (e.g., UI).
```
[the remark this answers is not in the document]
Comment (janisz) [reply README.md]: https://english.stackexchange.com/a/16173

File: docs/diagrams/cluster_bundle_creation_diagram.txt (hunk 1 above)
Comment (janisz) [reply docs/diagrams/cluster_bundle_creation_diagram.txt]: Fixed

File: README.md [hunk 3]
```diff
@@ -1,28 +1,82 @@
...
+
+In the past dcos-diagnostics was bundled with:
+
+* [dcos-checks](https://github.com/dcos/dcos-checks)
+* [dcos-checks-runner](https://github.com/dcos/dcos-check-runner)
+
+– see: https://github.com/dcos/dcos-diagnostics/pull/35
+In that time dcos-diagnostics was called `3dt`
+([DC/OS Distributed Diagnostics Tool](https://github.com/dcos/3dt/tree/master)).
+It was deprecated in Jun, 2017 but some references might still exists.
```
[the remark this answers is not in the document]
Comment (janisz) [reply README.md]: Fixed

File: README.md [hunk 4]
```diff
@@ -1,28 +1,82 @@
...
+1. Old serial API – single master calls every node for data. This API is deprecated and should be removed in DC/OS 2.2
+
+![deprecated cluster bundle creation diagram](docs/diagrams/deprecated_cluster_bundle_creation_diagram.png)
+
+2. New parallel API – single master schedules local bundle creation for every node in a cluster. Then master wait until
+nodes finishes bundles. Master downloads finished bundles and merge them into a single cluster bundle zip.
+
+![cluster bundle creation diagram](docs/diagrams/cluster_bundle_creation_diagram.png)
+
+Old API is faster for smaller clusters but it's slow for large cluster, so we recomend using only new API that's available
```
[the remark this answers is not in the document]
Comment (janisz) [reply README.md]: Fixed

File: README.md [hunk 5]
```diff
@@ -1,28 +1,82 @@
...
+You can see rendered version
+[here](https://temando.github.io/open-api-renderer/demo/?url=https://raw.githubusercontent.com/dcos/dcos-diagnostics/master/docs/api.yaml).
+There are two versions of bundle API.
+
+1. Old serial API – single master calls every node for data. This API is deprecated and should be removed in DC/OS 2.2
+
+![deprecated cluster bundle creation diagram](docs/diagrams/deprecated_cluster_bundle_creation_diagram.png)
+
+2. New parallel API – single master schedules local bundle creation for every node in a cluster. Then master wait until
+nodes finishes bundles. Master downloads finished bundles and merge them into a single cluster bundle zip.
```
[the remark this answers is not in the document]
Comment (janisz) [reply README.md]: Fixed

[the remark this answers is not in the document]
Comment (janisz) [reply README.md]: Fixed

File: README.md [hunk 6]
```diff
@@ -1,28 +1,82 @@
...
+(see: [api/rest/coordinator.go](https://github.com/dcos/dcos-diagnostics/blob/f719ffce0339f07f1be4a1ca24fb8b96fe94dff4/api/rest/coordinator.go#L361-L363)).
+
+The contents of the generated bundle are not stable over time and any internal or third party bundle analysis tooling
+should be programmed very defensively in this regard.
+See: [dcos-docs-site#2253](https://github.com/mesosphere/dcos-docs-site/pull/2253)
+
+#### API
+
+API documentation could be find in [docs](/docs) directory. It's using
+[OpenApi v3.0](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md)
```
[the remark this answers is not in the document]
Comment (janisz) [reply README.md]: Fixed

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Pull request merged**